lsp-completion.el

```diff
@@ -334,62 +334,49 @@ Return `nil' when fails to guess prefix."
 
 (defun lsp-completion--to-internal (items)
   "Convert ITEMS into internal form."
-  (--> items
-    (-map (-lambda ((item &as &CompletionItem
-                          :label
-                          :filter-text?
-                          :_emacsStartPoint start-point
-                          :score?))
-            `( :label ,(or (unless (lsp-falsy? filter-text?) filter-text?) label)
-               :item ,item
-               :start-point ,start-point
-               :score ,score?))
-          it)))
-
-(cl-defun lsp-completion--filter-candidates (items &key
-                                                   lsp-items
-                                                   markers
-                                                   prefix
-                                                   &allow-other-keys)
-  "List all possible completions in cached ITEMS with their prefixes.
-We can pass LSP-ITEMS, which will be used when there's no cache.
-The MARKERS and PREFIX value will be attached to each candidate."
+  (-map (-lambda ((item &as &CompletionItem
+                        :label
+                        :filter-text?
+                        :_emacsStartPoint start-point
+                        :score?))
+          `( :label ,(or filter-text? label)
```

Comment:
There's no guarantee that the `filter-text?` needs to be `nil`, it can be just an empty string which is still evaluated to `t`
See this issue https://github.com/emacs-lsp/lsp-mode/pull/2762

Reply:
ahhh this the result of putting down a branch and then coming back to it 2 weeks later forgetting what I've done haha

Reply:
This is the result of putting down the branch and then coming back 2 weeks later forgetting what I've done